### DISCUSSION TITLE
(.ts) TypeScript files are not correctly vendored

### DISCUSSION THREAD

**Discussion by xuyxu** [github-linguist/linguist#5323] [Classification] [answered]
Hi, thanks for developing this wonderful tool!

## Problem description
The dominant language turns into `TypeScript` recently, despite the fact that the [directory](https://github.com/alan-turing-institute/sktime/search?l=typescript) containing all typescript files is already marked as vendored in our [.gitattributes](https://github.com/alan-turing-institute/sktime/blob/main/.gitattributes).

### URL of the affected repository:
[sktime](https://github.com/alan-turing-institute/sktime)

* * *
We are pretty sure that several days ago, the dominant language of our repo was still `Python`. In addition, we have noticed that some other repos also have the same problem, such as [TS-Extrinsic-Regression](https://github.com/ChangWeiTan/TS-Extrinsic-Regression).

Will appreciate any help from you, Thanks!

cc @mloning

**Comment by lildude** [comment, marked as answer]
> despite the fact that the [directory](https://github.com/alan-turing-institute/sktime/search?l=typescript) containing all typescript files is already marked as vendored in our [.gitattributes](https://github.com/alan-turing-institute/sktime/blob/main/.gitattributes).

Almost. Your override isn't recursive so it's only marking the files within `sktime/datasets/data/` itself as vendored.

As per the [`.gitattributes` documentation](https://www.git-scm.com/docs/gitattributes):

> The rules by which the pattern matches paths are the same as in .gitignore files (see gitignore[5]), with a few exceptions:
>
> -  negative patterns are forbidden
> -  patterns that match a directory do not recursively match paths inside that directory (so using the trailing-slash path/ syntax is pointless in an attributes file; use path/** instead)

From the [`.gitignore` docs](https://www.git-scm.com/docs/gitignore):

> An asterisk "*" matches anything except a slash. 

All of your Typescript files are in sub-directories of that directory so you need to make your override recursive by using `**` instead of `*`.  I'm not sure why things would have been any different before as this has been the behaviour of git for a long time.

> In addition, we have noticed that some other repos also have the same problem, such as [TS-Extrinsic-Regression](https://github.com/ChangWeiTan/TS-Extrinsic-Regression).

That repo doesn't have any overrides so you're seeing the [default behaviour](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md#how-linguist-works). That repo has more bytes of code of TypeScript than Python.  In this case, you have ~40MB of TypeScript (thanks to the two massive files in `TS-Extrinsic-Regression/tree/master/data/Sample`) and only 100KB of Python in your entire repo. You'll need to implement an override if you don't want these to be counted.

**Reply by xuyxu** [reply]
Thanks @lildude, we will take your suggestions.

**Reply by lildude** [reply]
One thing to keep in mind, you will need to push other changes along with the `.gitattribute` changes as modifying this file alone is not enough to force a re-analysis of your repo.

**Reply by xuyxu** [reply]
Got it ;-)
